Tests/PackageRegistryTests/RegistryConfigurationTests.swift

```diff
@@ -0,0 +1,132 @@
+
+        configuration.defaultRegistry = .init(url: defaultRegistryBaseURL)
+        configuration.scopedRegistries["foo"] = .init(url: customRegistryBaseURL)
+        configuration.scopedRegistries["bar"] = .init(url: customRegistryBaseURL)
+
+        let encoded = try encoder.encode(configuration)
+        let decoded = try decoder.decode(RegistryConfiguration.self, from: encoded)
+
+        XCTAssertEqual(configuration, decoded)
+    }
+
+    func testDecodeEmptyConfiguration() throws {
+        let data = #"""
+        {
+            "registries": {},
+            "version": 1
+        }
+        """#.data(using: .utf8)!
+
+        let configuration = try decoder.decode(RegistryConfiguration.self, from: data)
+        XCTAssertNil(configuration.defaultRegistry)
+        XCTAssertEqual(configuration.scopedRegistries, [:])
+    }
+
+    func testDecodeExampleConfiguration() throws {
+        let data = #"""
+        {
+            "registries": {
+                "[default]": {
+                    "url": "\#(defaultRegistryBaseURL)"
+                },
+                "foo": {
+                    "url": "\#(customRegistryBaseURL)"
+                },
+                "bar": {
+                    "url": "\#(customRegistryBaseURL)"
+                },
+            },
+            "version": 1
+        }
+        """#.data(using: .utf8)!
+
+        let configuration = try decoder.decode(RegistryConfiguration.self, from: data)
+        XCTAssertEqual(configuration.defaultRegistry?.url, defaultRegistryBaseURL)
+        XCTAssertEqual(configuration.scopedRegistries["foo"]?.url, customRegistryBaseURL)
+        XCTAssertEqual(configuration.scopedRegistries["bar"]?.url, customRegistryBaseURL)
+    }
+
+    func testDecodeConfigurationWithInvalidRegistryKey() throws {
+        let data = #"""
+        {
+            "registries": {
+                0: "\#(customRegistryBaseURL)"
+            },
+            "version": 1
+        }
+        """#.data(using: .utf8)!
+
+        XCTAssertThrowsError(try decoder.decode(RegistryConfiguration.self, from: data))
+    }
+
+    func testDecodeConfigurationWithInvalidRegistryValue() throws {
+        let data = #"""
+        {
+            "registries": {
+                "[default]": "\#(customRegistryBaseURL)"
+            },
+            "version": 1
+        }
+        """#.data(using: .utf8)!
+
+        XCTAssertThrowsError(try decoder.decode(RegistryConfiguration.self, from: data))
+    }
+
+    func testDecodeConfigurationWithMissingVersion() throws {
+        let data = #"""
+        {
+            "registries": {}
+        }
+        """#.data(using: .utf8)!
+
+        XCTAssertThrowsError(try decoder.decode(RegistryConfiguration.self, from: data))
+    }
+
+    func testDecodeConfigurationWithInvalidVersion() throws {
+        let data = #"""
+        {
+            "registries": {},
+            "version": 999
+        }
+        """#.data(using: .utf8)!
```

Comment:
maybe we add Decoder extension in the testing module to take string, so we dont have to do the data dance everywhere?